app/calculators/FullTeamStatsCalculator.java

```diff
@@ -0,0 +1,71 @@
+package calculators;
+
+import interfaces.*;
+import interfaces.stats.FullTeamStatsCalculationResultDTO;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class FullTeamStatsCalculator implements StatsCalculator {
+    @Override
+    /** Calculate the entire team standings and returns a Data Transfer Object containing team standing info.
+     */
+    public FullTeamStatsCalculationResultDTO calculate(StatsGenerationRequestDTO statsGenerationRequest) {
+        FullTeamStatsCalculationResultDTO results = new FullTeamStatsCalculationResultDTO();
+        // initialize
+        List<TeamDTO> teams = statsGenerationRequest.getTeams();
+        List<MatchDTO> matches = statsGenerationRequest.getMatches();
+        // go through teams
+        // stats go in order of Win, Loss, Tie, Win pct, PPG, PAPG, Margin, TUH, PPTH, P/N, G/N, PPB
+        Map<String, List<MatchTeamStatsDTO>> teamStatsMap = new HashMap<>();
+        List<MatchTeamStatsDTO> stats = new ArrayList<>();
+        // fill stats with each team
+        for(MatchDTO curMatch: matches){
+            // calculate team 1
+            int numTeams = curMatch.getTeams().size();
+            // add and calculate results for each team of the match
+            for(int j = 0; j < numTeams; j++){
+                MatchTeamDTO curTeam = curMatch.getTeams().get(j);
```

Comment:
For consistency's sake, let's make this inner loop look like the outer for-loop that's using an iterator.